core/src/test/java/org/wildfly/channel/ChannelSessionTestCase.java

```diff
@@ -142,7 +143,7 @@ public void testResolveLatestMavenArtifact() throws Exception {
         File resolvedArtifactFile = mock(File.class);
 
         when(factory.create(any())).thenReturn(resolver);
-        when(resolver.getAllVersions("org.wildfly", "wildfly-ee-galleon-pack", null, null)).thenReturn(Set.of("25.0.0.Final", "25.0.1.Final"));
+        when(resolver.getAllVersions("org.wildfly", "wildfly-ee-galleon-pack", null, null)).thenReturn(new HashSet<>(Arrays.asList("25.0.0.Final", "25.0.1.Final")));
```

Comment:
@ivassile is the change from `Set.of` to `new HashSet(Arrays.asList)` needed?

Reply:
Yes, because Set.of returns immutable set. In my first implementation, I used the null value as indication that the blocklist is not present to avoid additional operations. Using emptySet() makes the code cleaner, but more calls will be made so I changed the tests to use mutable sets for the artifact versions because the new implementation calls removeAll() on the sets which causes UnsupportedOperationException.